qcodes/tests/test_autoloadable_channels.py

```diff
@@ -105,82 +128,94 @@ def __init__(
 
-        self._backend = TestInstrumentBackend()
+        self._backend = MockBackend()
 
         self.add_parameter(
             "channel_catalog",
             get_cmd=":INST:CHN:CAT",
         )
 
-    def write_raw(self, cmd: str):
-        return self._backend.send(cmd)
-
-    def ask_raw(self, cmd: str):
-        return self._backend.send(cmd)
+        channels = AutoLoadableChannelList(self, "channels", SimpleTestChannel)
+        self.add_submodule("channels", channels)
 
+    def write_raw(self, cmd: str)->None:
+        self._backend.send(cmd)
 
-class DummyInstrumentWithChannelList(TestInstrument):
-    def __init__(self, name):
-        super().__init__(name)
+    def ask_raw(self, cmd: str)->str:
+        return self._backend.send(cmd)
 
-        channels = AutoLoadableChannelList(self, "channels", SimpleTestChannel)
-        self.add_submodule("channels", channels)
 
+@pytest.fixture(scope='function')
+def dummy_instrument():
+    instrument = DummyInstrument("instrument")
+    yield instrument
+    instrument.close()
 
-def test_sanity():
 
-    instrument = TestInstrument("instrument")
-    channels = SimpleTestChannel.load_from_instrument(instrument)
+def test_sanity(dummy_instrument):
+    """
+    Test the basic functionality of the auto-loadable channels, without using
+    the auto-loadable channels list
+    """
+    # Test that we are able to discover instruments automatically
+    channels = SimpleTestChannel.load_from_instrument(dummy_instrument)
     assert len(channels) == 3
     assert channels[0].hello() == "Hello from channel 1"
     assert channels[1].hello() == "Hello from channel 2"
     assert channels[2].hello() == "Hello from channel 4"
-
-    new_channel_kwargs = SimpleTestChannel.get_new_instance_kwargs(instrument)
-    new_channel = SimpleTestChannel(instrument, **new_channel_kwargs)
-
-    with pytest.raises(RuntimeError):
-        new_channel.hello()  # We have only instantiated a channel class, we
-        # have yet to create it on the instrument
-
+    # Test that we can generate a new instance of the channels without conflicting names
+    new_channel_kwargs = SimpleTestChannel._get_new_instance_kwargs(dummy_instrument)
+    new_channel = SimpleTestChannel(dummy_instrument, **new_channel_kwargs)
+    # Instrument IO through the newly instantiated channel should raise an
+    # exception before actually creating the channel on the instrument
+    with pytest.raises(RuntimeError) as e1:
+        new_channel.hello()
+    assert e1.value.args[0] == "Object does not exist (anymore) on the instrument"
+    # After creating the channel we should be able to talk to the instrument.
     new_channel.create()
     assert new_channel.hello() == "Hello from channel 3"
-    new_channel.delete()
-
-    with pytest.raises(RuntimeError):
+    # Once we remove the channel we should no longer be able to talk to the instrument
+    new_channel.remove()
+    with pytest.raises(RuntimeError) as e2:
```

Comment:
you can use `pytest.raises(Exception, match="Object does not exist (anymore) on the instrument")` instead of working with `e2` explicitly.

Reply:
Oh I did not know that! Thanks!